src/sentry/eventstream/kafka/consumer.py

```diff
@@ -0,0 +1,295 @@
+    that partition. If the offsets recorded to that topic move
+    non-monotonically (due to an intentional offset rollback, for instance)
+    this consumer *may* consume up to the highest watermark point. (The
+    implementation here tries to pause consuming from the partition as soon as
+    possible, but this makes no explicit guarantees about that behavior.)
+    """
+    initial_offset_reset_strategies = {
+        'earliest': get_earliest_offset,
+        'latest': get_latest_offset,
+    }
+
+    def __init__(self, bootstrap_servers, consumer_group, commit_log_topic,
+                 synchronize_commit_group, initial_offset_reset='latest', on_commit=None):
+        self.bootstrap_servers = bootstrap_servers
+        self.consumer_group = consumer_group
+        self.commit_log_topic = commit_log_topic
+        self.synchronize_commit_group = synchronize_commit_group
+        self.initial_offset_reset = self.initial_offset_reset_strategies[initial_offset_reset]
+
+        self.__partition_state_manager = SynchronizedPartitionStateManager(
+            self.__on_partition_state_change)
+        self.__commit_log_consumer, self__commit_log_consumer_stop_request = self.__start_commit_log_consumer()
+
+        self.__positions = {}
+
+        def commit_callback(error, partitions):
+            if on_commit is not None:
+                return on_commit(error, partitions)
+
+        consumer_configuration = {
+            'bootstrap.servers': self.bootstrap_servers,
+            'group.id': self.consumer_group,
+            'enable.auto.commit': 'false',
+            'enable.auto.offset.store': 'true',
+            'enable.partition.eof': 'false',
+            'default.topic.config': {
+                'auto.offset.reset': 'error',
+            },
+            'on_commit': commit_callback,
+        }
+
+        self.__consumer = Consumer(consumer_configuration)
+
+    def __start_commit_log_consumer(self, timeout=None):
+        """
+        Starts running the commit log consumer.
+        """
+        stop_request_event = threading.Event()
+        start_event = threading.Event()
+        result = execute(
+            functools.partial(
+                run_commit_log_consumer,
+                bootstrap_servers=self.bootstrap_servers,
+                consumer_group='{}:sync:{}'.format(self.consumer_group, uuid.uuid1().hex),
+                commit_log_topic=self.commit_log_topic,
+                synchronize_commit_group=self.synchronize_commit_group,
+                partition_state_manager=self.__partition_state_manager,
+                start_event=start_event,
+                stop_request_event=stop_request_event,
+            ),
+        )
+        start_event.wait(timeout)
+        return result, stop_request_event
+
+    def __check_commit_log_consumer_running(self):
+        if not self.__commit_log_consumer.running():
+            try:
+                result = self.__commit_log_consumer.result(timeout=0)  # noqa
```

Comment:
What's this check for, did you intend to return or use `result` for anything?

Reply:
It's kind of weird but if result throws a non-TimeoutError exception, I wanted to raise that. If it doesn't throw anything, I wanted that to be logged to Sentry at least.